src/cmd/tools/crane.go

```diff
@@ -217,20 +224,23 @@ func pruneImages(_ *cobra.Command, _ []string) error {
 		return err
 	}
 
+	ctx, cancel := context.WithTimeout(context.Background(), cluster.DefaultTimeout)
+	defer cancel()
+
 	// Load the state
-	zarfState, err := c.LoadZarfState()
+	zarfState, err := c.LoadZarfState(ctx)
 	if err != nil {
 		return err
 	}
 
 	// Load the currently deployed packages
-	zarfPackages, errs := c.GetDeployedZarfPackages()
+	zarfPackages, errs := c.GetDeployedZarfPackages(ctx)
 	if len(errs) > 0 {
 		return lang.ErrUnableToGetPackages
 	}
 
 	// Set up a tunnel to the registry if applicable
-	registryEndpoint, tunnel, err := c.ConnectToZarfRegistryEndpoint(zarfState.RegistryInfo)
+	registryEndpoint, tunnel, err := c.ConnectToZarfRegistryEndpoint(ctx, zarfState.RegistryInfo)
```

Comment:
I feel like it's very unlikely a few secrets + a svc times this out at 30s but it still gives me a little pause since people might be in uber slow environments. Especially since there's no override atm and we're combining multiple actions under the same timeout.

Are we able to mimic a situation locally where a timeout would be beneficial? I'm assuming if we can't connect to the cluster we will error pretty quick, but I wonder if there's a state our users could get stuck in. That would convince me we should add the timeouts.

Reply:
fixed